core/src/main/java/org/radargun/stages/test/SchedulingSelector.java

```diff
@@ -0,0 +1,131 @@
+package org.radargun.stages.test;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicIntegerArray;
+import java.util.concurrent.atomic.AtomicLongArray;
+
+import org.radargun.logging.Log;
+import org.radargun.logging.LogFactory;
+import org.radargun.utils.Projections;
+import org.radargun.utils.TimeService;
+
+/**
+ * Based on provided frequency, returns matching invocation from {@link #next()} or blocks the thread calling it.
+ *
+ * @author Radim Vansa &lt;rvansa@redhat.com&gt;
+ */
+public class SchedulingSelector<T> {
+   private final static Log log = LogFactory.getLog(SchedulingSelector.class);
+   private final static boolean trace = log.isTraceEnabled();
+   private final int[] invocations;
+   private final long[] intervals;
+   private final T[] operations;
+   private final int numOperations;
```

Comment:
Is it worth introducing separate variable for numOperations when it can be derived from operations array?